src/main/java/core/basesyntax/interfaces/WriteToFileImpl.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.interfaces;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class WriteToFileImpl implements WriteToFile {
+
+    @Override
+    public void write(String filePath, List<String> report) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            for (String line : report) {
+                writer.write(line);
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("There is a problem with a file!", e);
```

The catch block rethrows the IOException with the cause (good), but the message is generic: "There is a problem with a file!" The checklist requires rethrowing with an informative message that includes context. Quote: "Do not ignore exceptions: on failures (e.g., file not found, I/O errors), rethrow as RuntimeException with informative messages and the original exception." Include the `filePath` in the message (e.g., "Can't write report to file: " + filePath, e).